### PR TITLE
Issue384 - autopublish

### LINKED ISSUE #384: [FEAT] -  automatically publishes messages sent in announcement channels
### What feature would you like to see?

like [https://github.com/Benricheson101/Publisher](https://github.com/Benricheson101/Publisher)

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by MoOaAaa**
# Pull Request

## Description

I tried including a publish command, available only on server that disposes of news channels (community servers).
I added a field in the server settings, where you can set the announcement channel you want the bot to publish in.
You can set the channel with the command `set autoPublishChannel <yourChannel>`. The channel must exist to be set (based on the same test as in set.js).
The command is called with `publish <message ID/link>`.

There is a large room for improvement, but as I am really new to discord bots, please guide me to fix mistakes or add new functionality. 
The code is based on the link furnished in #384 ([https://github.com/Benricheson101/Publisher](https://github.com/Benricheson101/Publisher)).

## Related Issue

Closes #384 

## Version
Did you remember to update the version number in the package.json file?

At the time of the pull request, no changes can be seen between the main code and my fork.
- [ ] For a major restructuring, or large changes please bump the the major version number. (ex: 1.x.x)
- [x] For a minor change, bump the minor version number. (ex: x.1.x)
- [ ] For a bug fix, bump the patch version number. (ex: x.x.1)

## Documentation

For _any_ version updates, please verify if the [documentation page](https://docs.beccalyria.com?utm_source=github&utm_medium=pr-template) needs an update. If it does, please [create an issue there](https://github.com/BeccaLyria/discord-documentation/issues/new?assignees=nhcarrigan&labels=%F0%9F%9A%A6+status%3A+awaiting+triage&template=update.md&title=%5BUPDATE%5D) to ensure it is not forgotten.

- [x] My contribution does NOT require a documentation update.
- [ ] My contribution DOES require a documentation update.

### CODE REVIEW

**Comment by j-dogcoder** [comment]
Hey @MoOaAaa, this is a minor bump to the version. Please check the minor box, and bump the package.json accordingly.

File: src/publish.js [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+ * The channel must be setup when the bot arrives on the server, and the bot need to be able to manage messages.
+ */
+
+const config = require("./config");
+const { getSettings }= require("./utils/functions");
+
+const CLOCK = '⏲️';
+const MEGAPHONE = '📢';
+const EXCLAMATION = '❗';
+const ROTATING_LIGHT = '🚨';
```
Comment (j-dogcoder) [inline src/publish.js]: Hey! Emojis should be stored in the config.js file please.

File: src/publish.js [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+    if(checkPermission(message,)) return reaction(message, ROTATING_LIGHT);
+
+    //Check if there is an argument to the command
+    if(!args) return message.channel.send(`:x: You must precise the link or id of the message you want to publish. *Exemple : \`${settings.prefix}publish [link-to-your-message]*`);
+
+    //Get the message from the link or ID provided
+    const [msg, channel] = await getMessage(message, args[0]);
+
+    //Check if the message is in the discord server
+    if  (msg.guild.id !== message.guild.id) return;
```
Comment (j-dogcoder) [inline src/publish.js]: Not needed, you can delete these 2 lines.

File: src/publish.js [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+exports.run = async (client, message, args, level) =>{
+    const settings = message.settings = getSettings(message.guild);
+    const autoPublishChannel = settings["autoPublishChannel"];
+
+
+    //Check if the you have persmission to manage messages and send messages
+    if(checkPermission(message,)) return reaction(message, ROTATING_LIGHT);
+
+    //Check if there is an argument to the command
+    if(!args) return message.channel.send(`:x: You must precise the link or id of the message you want to publish. *Exemple : \`${settings.prefix}publish [link-to-your-message]*`);
```
Comment (j-dogcoder) [inline src/publish.js]: *Grammer*

File: src/publish.js [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+
+    //Get the message from the link or ID provided
+    const [msg, channel] = await getMessage(message, args[0]);
+
+    //Check if the message is in the discord server
+    if  (msg.guild.id !== message.guild.id) return;
+
+    //Check if the message exist
+    if (!msg){
+        return message.channel.send(":x: That message doesn't exist. Check if you messsage ID is right or try using message links instead.");
```
Comment (j-dogcoder) [inline src/publish.js]: *Grammer*

File: src/publish.js [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+    if (channel.type !== 'news'){
+        return message.channel.send(":x: The message was not sent in an announcement channel.");
+    }
+
+    //Check if the bot has the permission to send messages
+    if  (
+        !msg.guild.me.permissionsIn(channel).has('MANAGE_MESSAGES') ||
+        !msg.guild.me.permissionsIn(channel).has('SEND_MESSAGES')
+    ){
+        return await message.channel.send(":x: I do not have the permissions required to publish messages. Please make sure I can bot `MANAGE_MESSAGE` and `SEND_MESSAGES`.")
```
Comment (j-dogcoder) [inline src/publish.js]: *Grammer*

**Review by jaspermayone — CHANGES_REQUESTED** [review]

**Comment by j-dogcoder** [comment]
Hey @MoOaAaa! I left some feedback.

**Review by jaspermayone — APPROVED** [review]

**Pull request merged**